waddrmgr/manager_test.go

```diff
@@ -1016,6 +1017,32 @@ func testImportScript(tc *testContext) bool {
 	return true
 }
 
+// testMarkUsed ensures used addresses are flagged as such.
+func testMarkUsed(tc *testContext) bool {
+	expectedAddr1 := expectedAddr{
+		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
+		used:        true,
+	}
+	prefix := "MarkUsed"
+	chainParams := tc.manager.ChainParams()
+	addrHash := expectedAddr1.addressHash
+	addr, err := btcutil.NewAddressPubKeyHash(addrHash, chainParams)
+
```

Comment:
Please add a test here to make sure the address is not marked used before it's marked as used below.  This will also ensure that the cached address is properly updated.